source/MaterialXCore/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Generated.h.in ${CMAKE_CURRENT_SOURCE_DIR}/Generated.h)
+configure_file(${CMAKE_CURRENT_LIST_DIR}/Generated.h.in ${CMAKE_CURRENT_BINARY_DIR}/Generated.h)
```

Comment:
I can see the need for the generated header to be in CMAKE_CURRENT_BINARY_DIR, and that part of the change seems like a clear improvement.

Can you provide some clarity on the motivation for referencing source files in CMAKE_CURRENT_LIST_DIR rather than CMAKE_CURRENT_SOURCE_DIR?  I'm not against this idea, but I want to make sure that I understand the logic behind the change.

Reply:
In this context it does not matter that much and both are equivalent, so I will probably revert this change to keep the diff minimal. 

My habit though it to use `CMAKE_CURRENT_LIST_DIR` when I can as it always refers to the CMake file currently being processed as the other one may not. 

Here is a page with a bit more details and examples if you want to dig this : https://cgold.readthedocs.io/en/latest/tutorials/cmake-sources/common.html#cmake-current-list-dir-vs-cmake-current-source-dir